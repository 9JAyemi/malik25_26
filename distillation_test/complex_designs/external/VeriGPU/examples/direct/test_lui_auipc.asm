# LUI, AUIPC

lui x1, 0xabcde
outr x1
addi x1, x1, 0x321
outr x1

auipc x1, 0x111
outr x1

; LI
li x2, 0xabcde321
outr x2

li x2, 0xabcdefff
outr x2

li x2, 0xabcde000
outr x2

addi x1, x0, 10
outr x1

addi x1, x0, -10
outr x1

halt
