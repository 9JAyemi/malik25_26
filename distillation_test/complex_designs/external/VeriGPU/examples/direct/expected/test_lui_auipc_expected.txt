out 0 10101011110011011110000000000000 abcde000 2882396160
out 1 10101011110011011110001100100001 abcde321 2882396961
out 2 00000000000100010001000010100000 001110a0 1118368
out 3 10101011110011011110001100100001 abcde321 2882396961
out 4 10101011110011011110111111111111 abcdefff 2882400255
out 5 10101011110011011110000000000000 abcde000 2882396160
out 6 00000000000000000000000000001010 0000000a 10
out 7 11111111111111111111111111110110 fffffff6 4294967286
